### PR TITLE
Property-based tests for `sets`, `ordsets` and `gb_sets`

### PR DESCRIPTION
**Pull request by Maria-12648430**
This PR provides property-based tests for the `sets` module.

It does not test `sets:new/0,1`, since there isn't much that could be property-tested there.

### CODE REVIEW

**Comment by Maria-12648430** [comment]
If this gets accepted and merged before #7232 and/or #7183, this PR should be updated to include the functions introduced there.

**Comment by juhlig** [comment]
> If #7232 and/or #7183 get accepted and merged earlier, this PR should be updated to include the functions introduced there.

The equality check functions from #7232 would be pretty handy in this PR I guess 🙂

File: lib/stdlib/test/property_test/sets_prop.erl [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+%% |    | S1   +-------------+ | |
+%% |    |      | S2          | | |
+%% |    | +----+-----------+ | | |
+%% |    | | S3 | +-------+ | | | |
+%% |    | |    | | Empty | | | | |
+%% |    | |    | +-------+ | | | |
+%% |    | |    +-----------+-+ | |
+%% |    | +----------------+   | |
+%% |    +----------------------+ |
+%% +-----------------------------+
```
Comment (TD5) [inline lib/stdlib/test/property_test/sets_prop.erl:170]: I can't help but comment to say that this is excellent use of ASCII art!

**Comment by Maria-12648430** [comment]
The last commit does some cleanup (ordering/grouping of functions, comments) and adds another property to test sequences of modifying operations.

**Comment by Maria-12648430** [comment]
On a side note, this suite makes heavy use of the `any()` generator (explicit and implicit via the `list()` generator), which contains the `atom()` generator. The `atom()` generator produces atoms at random, which can become a problem re the atom limit.

**Comment by Maria-12648430** [comment]
I marked the PR as WIP, as it still needs some touching up :)

**Comment by Maria-12648430** [comment]
Last commit adds tests for the new functions `map/2`, `filtermap/2` and `is_equal/2`.

**Comment by Maria-12648430** [comment]
As it stands, this suite alone generates ~800,000 new atoms. Waiting for #7364 which should remedy this.

**Comment by Maria-12648430** [comment]
The last commit uses the atomlimit-safe generators of `ct_proper_ext`, provided by #7364.

**Comment by bjorng** [comment]
Please squash your commits.

**Comment by Maria-12648430** [comment]
> Please squash your commits.

Done :wink:

**Comment by bjorng** [comment]
Thanks! Added to our daily builds.

**Comment by bjorng** [comment]
I will now go on vacation and probably be back in about four weeks. @jhogberg will take care of merging this pull request if no problems turn up in our daily builds.

**Comment by Maria-12648430** [comment]
> I will now go on vacation

Enjoy :wave:

> @jhogberg will take care of merging this pull request if no problems turn up in our daily builds.

And hi John :handshake:

**Comment by Maria-12648430** [comment]
Rebased on current master to make use of #7495.

**Comment by Maria-12648430** [comment]
@jhogberg @bjorng I wonder, what is the status of this PR? I just now noticed that the testing label was removed shortly after I pushed the last update :thinking:

On another note, I realized that the tests in this suite cover only the functions that the three set implementations have in common, but `gb_sets` has some additional functions that the other two don't have, like `singleton`, `delete`, `smallest`, `take_largest`, `iterator` and some more. Should I add them to this suite (like in `sets_SUITE`) or put up another property test suite for the additional functions of `gb_sets`), what do you think?

**Comment by jhogberg** [comment]
> @jhogberg @bjorng I wonder, what is the status of this PR? I just now noticed that the testing label was removed shortly after I pushed the last update 🤔

The status is that it fell between the cracks. :sweat_smile: 

I'll merge this in the next few days if the tests pass. :)

> On another note, I realized that the tests in this suite cover only the functions that the three set implementations have in common, but gb_sets has some additional functions that the other two don't have, like singleton, delete, smallest, take_largest, iterator and some more. Should I add them to this suite (like in sets_SUITE) or put up another property test suite for the additional functions of gb_sets), what do you think?

Feel free to do the latter in a separate PR.

**Comment by Maria-12648430** [comment]
> The status is that it fell between the cracks. :sweat_smile:

Ah, no problem :wink:

> Feel free to do the latter in a separate PR.

Will do :+1:

**Comment by Maria-12648430** [comment]
@bjorng so, with @jhogberg going on vacation, it seems that this PR has come back to ~haunt~ you :grin:

I noticed that I forgot the copyright header and added it in the last commit.

**Comment by bjorng** [comment]
Thanks for your pull request.

**Pull request merged**